src/inference/enumerate.ad.js

```diff
@@ -47,6 +57,17 @@ module.exports = function(env) {
     return this.wpplFn(_.clone(this.store), env.exit, this.a);
   };
 
+  // Error function for error handling
+  // this.throwOnError is true: directly throw error
+  // this.throwOnError is false: return error (string) as infer result
+  Enumerate.prototype.error = function(errType) {
```

Comment:
If `err` is an `Error` instance, you can still access the message as `err.message`, so this should be a small change to `header.wppl`, right?